src/main/java/net/greatstart/controllers/PasswordResetController.java

```diff
@@ -0,0 +1,144 @@
+import net.greatstart.services.UserService;
+import net.greatstart.validators.PasswordValidationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.MessageSource;
+import org.springframework.core.env.Environment;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Controller;
+import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+import java.util.Locale;
+import java.util.UUID;
+
+@Controller
+public class PasswordResetController {
+
+    private PasswordEncoder passwordEncoder;
+    private MessageSource messages;
+    private JavaMailSender mailSender;
+    private UserService userService;
+    private SecurityService securityService;
+    private PasswordValidationService validationService;
+    private Environment env;
+
+    @Autowired
+    public PasswordResetController(@Qualifier("messageSource") MessageSource messages, Environment env, JavaMailSender
+        mailSender, UserService userService, SecurityService securityService, PasswordEncoder passwordEncoder,
+                                   PasswordValidationService validationService) {
+        this.messages = messages;
+        this.env = env;
+        this.mailSender = mailSender;
+        this.userService = userService;
+        this.securityService = securityService;
+        this.passwordEncoder = passwordEncoder;
+        this.validationService = validationService;
+    }
+
+    @GetMapping(value = "/user/resetPassword")
+    public ModelAndView showForgotPassForm() {
+        ModelAndView model = new ModelAndView("user/forgotPassword");
+        model.addObject("user", new DtoUser());
+        return model;
+    }
+
+    @PostMapping(value = "/user/resetPassword")
+    public ModelAndView processEmail(HttpServletRequest request, @Valid @ModelAttribute("user") DtoUser dtoUser, Errors
+        errors) {
+        ModelAndView model = new ModelAndView("user/forgotPassword");
+        if (errors.hasErrors()) {
+            return model;
+        }
+        User user = userService.getUserByEmail(dtoUser.getEmail());
+        if (user == null) {
+            model.addObject("message", messages.getMessage("user.notFound", null, request.getLocale()));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "message" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)